File: cs/src/main/java/com/brightk/cs/core/annotation/Interceptor.java
package com.brightk.cs.core.annotation;


import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target({ElementType.TYPE})
@Retention(RetentionPolicy.RUNTIME)
public @interface Interceptor {
    String name();

    /**
     * 优先级
     * @return
     */
    int priority() default 0;
}

File: cs/src/main/java/com/brightk/cs/core/annotation/Interceptor.kt
package com.brightk.cs.core.annotation


@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class Interceptor(
    val name: String,
    /**
     * 优先级
     * @return
     */
    val priority: Int = 0
)
